### PR TITLE
JS2-WEEK2

### PR DESCRIPTION
**Pull request by Merongk**
project for the 2nd week of js2

### CODE REVIEW

**Review by vdaguenet — COMMENTED** [review]
Merge the week 1 PR and implement the changes in the same files first. Then I'll review the week 2.

**Review by Merongk — COMMENTED** [review]
new changes

File: App.js
Comment (vdaguenet) [inline App.js:1]: This file is not used. You can delete it.

File: index.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        .grid-item span {
+            display: block;
+            font-size: 16px;
+            color: #555;
+        }
+    </style>
+</head>
+<body>
+    <h1>Currency Rates</h1>
+    <div class="search-container">
```
Comment (vdaguenet) [inline index.html:127]: This could be a `form` element

File: index.html [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        <input type="text" id="searchFrom" placeholder="From Currency">
+        <input type="text" id="searchTo" placeholder="To Currency">
+        <button onclick="searchCurrencyRate()">Search</button>
```
Comment (vdaguenet) [inline index.html:130]: Prefer binding events with `addEventListener` in JS instead of HTML attributes.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    </div>
+    <div id="currencyGrid" class="grid"></div>
+
+    <script>
```
Comment (vdaguenet) [inline index.html:134]: Where did the week 1 features go? The goal of the project is to implement new features every week and get a fully functional currency converter at the end of JavaScript module 3.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                item.innerHTML = `
+                    <strong>${rate.from} to ${rate.to}</strong>
+                    <span>Rate: ${rate.rate}</span>
+                `;
+                grid.appendChild(item);
+            });
+        }
+
+        function searchCurrencyRate() {
+            const from = document.getElementById('searchFrom').value.toUpperCase();
```
Comment (vdaguenet) [inline index.html:160]:
```suggestion
            const from = document.getElementById('searchFrom').value.toUpperCase().trim();
```
As mentioned [in the last PR](https://github.com/Merongk/js-projects-currency-converter/pull/2#discussion_r1675568941).

File: index.html [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            const to = document.getElementById('searchTo').value.toUpperCase();
+            const filteredRates = currencyRates.filter(rate => rate.from === from && rate.to === to);
```
Comment (vdaguenet) [inline index.html:162]: This is a good start for finding a rate when the `from` and `to` currencies are provided. However, the code must also work when only one currency is provided. Have a look at my messages on Slack. You can either make a search function with two arguments like [this](https://hackyourfuture-cph.slack.com/archives/C07AT59R9QR/p1720790626662359) or have only one argument like [this](https://hackyourfuture-cph.slack.com/archives/C07AT59R9QR/p1720794065032809).

**Pull request merged**